Readme.md

```diff
@@ -29,11 +30,47 @@ func main() {
     // Instantiates a client to send messages to the dreamdata API.
     client := analytics.New(os.Getenv("DREAMDATA_WRITE_KEY"))
 
-    // Enqueues a track event that will be sent asynchronously.
-    client.Enqueue(analytics.Track{
-        UserId: "test-user",
-        Event:  "test-snippet",
-    })
+	errGroup := new(errgroup.Group)
+
+	errGroup.Go(c.Enqueue(analytics.Identify{
+		UserId: "019mr8mf4r",
+		Traits: map[string]interface{}{
+			"email": "user@dreamdata.io",
+			"name":  "user",
+			"age":   25,
+		},
+	}))
+
+	errGroup.Go(c.Enqueue(analytics.Track{
+		UserId: "019mr8mf4r",
+		Event:  "Song Played",
+		Properties: map[string]interface{}{
+			"name":   "My song",
+			"artist": "My artist",
+		},
+	}))
+
+	errGroup(c.Enqueue(analytics.Identify{
+		UserId: "971mj8mk7p",
+		Traits: map[string]interface{}{
+			"email": "user2@dreamdata.io",
+			"name":  "user2",
+			"age":   26,
+		},
+	}))
+
+	errGroup(c.Enqueue(analytics.Track{
+		UserId: "971mj8mk7p",
+		Event:  "Song Played",
+		Properties: map[string]interface{}{
+			"name":   "My song",
+			"artist": "My artist",
+		},
+	}))
+
+	if err := errGroup.Wait(); err == nil {
+		fmt.Printf("Successfully enqueued messages")
+	}
```

Comment:
since a live system would never use `errGroup` for dynamic events, simplify the example to something that handles the errors individually. An error will really only occur if/when Client.Close() is called prior to calling enqueue, so it is also fine to ignore them like segment is recommending [here](https://segment.com/docs/connections/sources/catalog/libraries/server/go/).

Also, you might want to show how to do the properties in a type-safe manner as seen [here](https://segment.com/docs/connections/sources/catalog/libraries/server/go/#whats-new-in-v3)

Reply:
Oh, I didn't see the generic `Set()` method for custom fields. Fixed all in commit d7391efc7bff37fac20ec334f2120615217ab48e